### PR TITLE
added support to include recursive fetch spec

### PR DESCRIPTION
**Pull request by srksag**

### CODE REVIEW

File: lib/knex/query.ts [hunk 1]
```diff
@@ -238,7 +239,16 @@ module.exports = function (PersistObjectTemplate) {
                 var type = defineProperty.type;
                 var of = defineProperty.of;
                 var cascadeFetch = (cascade && typeof(cascade[prop] != 'undefined')) ? cascade[prop] : null;
-
+                if (cascadeFetch && cascadeFetch.fetch) {
```
Comment (nsaenz7) [inline lib/knex/query.ts]: is there any limit on how deep we let the recursion go? e.g. if this recurses 100 times are we ok with that? what does that query end up looking like?

Comment (nsaenz7) [reply lib/knex/query.ts]: is there any way somebody could accidentally blow things up by setting a recursive fetch spec on an object with a circular reference?

Comment (sshankar19) [reply lib/knex/query.ts]: I had the same concern as nick here, how do we limit this incase of gigantic fetches?

Comment (srksag) [reply lib/knex/query.ts]: I've created https://massmutual.atlassian.net/browse/HL-16058..
Can I get approvals on this PR

**Comment by nsaenz7** [comment]
can we update whatever documentation we have in confluence to reflect this change?

**Pull request merged**